### ISSUE TITLE
Running with `-H:-RunReachabilityHandlersConcurrently` results in non-deterministic failures

### ISSUE THREAD

**Issue by zakkak** [oracle/graal#6904]
**Describe the issue**

When running native-image with `-H:-RunReachabilityHandlersConcurrently` it crashes non-deterministically.

Issue first reported in https://github.com/apache/camel-quarkus/issues/5006
The flag seems to be added by the [com.google.api:gax-grpc ](https://github.com/googleapis/gax-java/blob/32e99616b3112f89121f7c5a0b4e500d1e693846/gax-grpc/src/main/resources/META-INF/native-image/com.google.api/gax-grpc/native-image.properties#L14) dependency.

@fniephaus this is probably a duplicate of #5868

**Steps to reproduce the issue**
```
native-image --no-fallback -H:-RunReachabilityHandlersConcurrently HelloWorld
```

It might require multiple runs untill a crash.

**Describe GraalVM and your environment:**
 - GraalVM version: **GraalVM CE 17.0.7+7.1**
 - JDK major version: **17**
 - OS: **Fedora 38**
 - Architecture: **AMD64**

**More details**

I tried reproducing the issue on GraalVM 22.3 with no success.

```
========================================================================================================================
GraalVM Native Image: Generating 'helloworld' (executable)...
========================================================================================================================
[1/8] Initializing...                                                                                    (1.5s @ 0.16GB)
 Java version: 17.0.7+7, vendor version: GraalVM CE 17.0.7+7.1
 Graal compiler: optimization level: 2, target machine: x86-64-v3
 C compiler: gcc (redhat, x86_64, 13.1.1)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
[2/8] Performing analysis...  []                                                                         (1.9s @ 0.22GB)
   2,766 (75.74%) of  3,652 types reachable
   3,384 (50.73%) of  6,671 fields reachable
  12,536 (55.27%) of 22,680 methods reachable
     859 types,     0 fields, and   159 methods registered for reflection

------------------------------------------------------------------------------------------------------------------------
                        0.2s (5.8% of total time) in 19 GCs | Peak RSS: 0.98GB | CPU load: 12.42
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /home/zakkak/code/tmp/helloworld/svm_err_b_20230628T164422.007_pid665130.md (build_info)
========================================================================================================================
Failed generating 'helloworld' after 3.4s.

The build process encountered an unexpected error:

> java.lang.NullPointerException: Cannot invoke "java.util.Set.iterator()" because "triggers" is null

Please inspect the generated error report at:
/home/zakkak/code/tmp/helloworld/svm_err_b_20230628T164422.007_pid665130.md

If you are unable to resolve this problem, please file an issue with the error report at:
https://graalvm.org/support
```

**Comment by fniephaus** [comment]
> this is probably a duplicate of https://github.com/oracle/graal/issues/5868

Yes, very likely. Can we close it as duplicate?

**Comment by zakkak** [comment]
> Yes, very likely. Can we close it as duplicate?

Sure, I just find this description more clear, and with a simpler reproducer. If you close it please consider copy pasting the information.

**Comment by oubidar-Abderrahim** [comment]
Closed as duplicate of #5868
